src/Microsoft.Health.Fhir.Core/Features/Definition/SearchableSearchParameterDefinitionManager.cs

```diff
@@ -37,7 +37,7 @@ public SearchableSearchParameterDefinitionManager(SearchParameterDefinitionManag
 
         public IEnumerable<SearchParameterInfo> GetSearchParameters(string resourceType)
         {
-            if (_fhirReqeustContextAccessor.FhirRequestContext.IncludePartiallyIndexedSearchParams)
+            if (_fhirReqeustContextAccessor.FhirRequestContext == null || _fhirReqeustContextAccessor.FhirRequestContext.IncludePartiallyIndexedSearchParams)
```

Comment:
>if (_fhirReqeustContextAccessor.FhirRequestContext == null [](start = 12, length = 58)

nit: the null check here seems a little out-of-the-ordinary since we don't usually expect `FhirRequestContext` to be null. Can we add a comment here on when this scenario might happen?

Reply:
Also when it is null, should we be returning the parameters that are `IsSearchable` instead of `IsSupported`? For member-match, I think we would only want to consider parameters that are currently searchable

---
In reply to: [622629071](https://github.com/microsoft/fhir-server/pull/1843#discussion_r622629071) [](ancestors = 622629071)

Reply:
I agree.
It also has nothing to do with member-match, btw. It has to do with starting server and reading search parameters and when you do that, you don't have context at all.